### PR TITLE
Add Dart support

### PR DESCRIPTION
**Pull request by osa1**

### CODE REVIEW

**Comment by Wilfred** [comment]
Looks great! Would you mind adding a test file in `sample_files/` too? Once that's sorted, and CI is happy, I'm happy to merge this :)

**Comment by osa1** [comment]
@Wilfred Done.

I'm not sure if I understand this repo's organization. Am I right that in `vendor/` you don't add submodules, but copy contents of other repos?

There seems to be some issues in the `.git` of the repo. For example, if I run this in master branch:

```
 $ git submodule update --init
fatal: No url found for submodule path 'vendor/tree-sitter-bash/examples/bash-it' in .gitmodules
```

In any case, I believe this PR should be working. I've already started using it.

**Comment by Wilfred** [comment]
It's using `git subtree`, not `git submodule`. There's example commands here: https://difftastic.wilfred.me.uk/upstream_parsers.html#adding-a-new-parser

I wouldn't expect `git submodule` commands to do anything useful in difftastic. I think the error there is because the bash grammar happens to contain a submodule: https://github.com/tree-sitter/tree-sitter-bash/tree/master/examples

**Comment by osa1** [comment]
Is there anything you want me to do here @Wilfred ?

File: src/tree_sitter_parser.rs [hunk 1]
```diff
@@ -190,6 +191,20 @@ pub fn from_language(language: guess::Language) -> TreeSitterConfig {
                 .unwrap(),
             }
         }
+        Dart => {
+            let language = unsafe { tree_sitter_dart() };
+            TreeSitterConfig {
+                name: "CSS",
```
Comment (Wilfred) [inline src/tree_sitter_parser.rs:197]: Looks like a copy-paste issue here.

**Comment by Wilfred** [comment]
Merged (with a few tiny tweaks), thanks :)

**Comment by Wilfred** [comment]
Upstream highlighting bug: https://github.com/UserNobody14/tree-sitter-dart/issues/24

**Pull request closed without merging**